File: ios/Cache/Operation/Operation.swift
//
// Copyright © 2023 Nevis Security AG. All rights reserved.
//

import NevisMobileAuthentication

typealias ErrorHandler = (MobileAuthenticationClientError) -> ()

protocol Operation {
	var state: OperationState { get }
	var onError: ErrorHandler { get }

	func cancel()
}

File: src/operations/Operation.ts
/**
 * Copyright © 2023 Nevis Security AG. All rights reserved.
 */

/**
 * The generic interface representing an operation returned by {@link Operations}.
 */
export abstract class Operation {
	/**
	 * Executes the operation asynchronously.
	 */
	abstract execute(): Promise<void>;
}
